### PR TITLE
[Telink] Fix restart BLE adv after basic commissioning fail

### PR DESCRIPTION
**Pull request by serhiiSalamakha**
To restart the BLE advertising after the basic commissioning failed to need to restore the default values to the setting partition after erasing. 
For this need to reboot the system only after a problem during the basic commissioning mode.
The setting partition will be set during the system initialization.

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Comment by s07641069** [comment]
@bzbarsky-apple Could you perform PR review?

**Comment by andy31415** [comment]
> @bzbarsky-apple Could you perform PR review?

@s07641069 Boris is in vacation until Aug 27. Could get another reviewer (ask on slack) or if nobody else availabe I can fast track this as platform-specific fix

**Comment by s07641069** [comment]
> > @bzbarsky-apple Could you perform PR review?
> 
> 
> 
> @s07641069 Boris is in vacation until Aug 27. Could get another reviewer (ask on slack) or if nobody else availabe I can fast track this as platform-specific fix

No problem let's merge.
@bzbarsky-apple can give additional feedback after merge.

File: examples/platform/telink/common/src/AppTaskCommon.cpp
Comment (bzbarsky-apple) [inline examples/platform/telink/common/src/AppTaskCommon.cpp:161]: Why does basic vs enhanced matter here?

If commissioner A opens a basic commissioning window, commissioner B is trying to commission, and commissioner A removes itself during that time... why would we want to reset mid-commissioning?

**Pull request merged**
